src/Symfony/Component/Security/Http/AccessToken/OidcAccessTokenHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\Exception\InvalidOidcUserException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * The token handler validates and retrieves the user identifier from an OIDC server using the token.
+ *
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class OidcAccessTokenHandler implements AccessTokenHandlerInterface
+{
+    private HttpClientInterface $oidcClient;
```

Comment:
Since SF6.3 requires PHP 8.1+, the following could be used:

```php
    public function __construct(
        private readonly HttpClientInterface $oidcClient,
        private readonly LoggerInterface $logger = null,
        private readonly string $userProperty = 'email'
    )
    {}
```